Runtime

snoe925 opened the pull request:
Add ```Tcl_Interp * Tcl_CreateInterpWithStubs(const char *version, int exact);``` to create TCL
interpreters only using the cpptcl/cpptcl.h header.
This requires a new library cpptcl_runtime if you use this function to create a TCL interpreter in an executable.

The cpptcl library expects users to have USE_TCL_STUBS and to rely only on the stubs library.  This is the desired method for extensions. If you create an interpreter, then the function Tcl_CreateInterpWithStubs can be used to make the correct calls on the TCL library for setup.

Commit:
Add cpptcl_runtime

It can be difficult to get the headers, tcl.h and cpptcl.h
in correct order and settings when creating the first interpreter.
The new global function
   Tcl_CreateInterpWithStubs(const char *version, int exact)
make this easier as the library, cpptcl_runtime, is compiled without
stubs.

cpptcl end users can now get tcl.h from cpptcl.h.  They can also
create interpreters by changing

Tcl_CreateInterp() -> Tcl_CreateInterpWithStubs()

and they can remote the initialization calls.

Commit:
Add Tcl_CreateInterpWithStubs

Add a helper function for TCL interpreter creation.

conej730 commented on CMakeLists.txt:
```diff
@@ -73,5 +73,8 @@ install(TARGETS cpptcl_static ARCHIVE DESTINATION lib)
 
+add_library(cpptcl_runtime SHARED ${cpptcl_SOURCE_DIR}/cpptcl_runtime.c)
+target_link_libraries(cpptcl_runtime ${TCL_LIBRARY})
+
```
Can we get a static library too?  Firehose is using all statics so it can just create a "fat" binary.
```suggestion
add_library(cpptcl_runtime_static STATIC ${cpptcl_SOURCE_DIR}/cpptcl_runtime.c)
target_link_libraries(cpp_runtime_static ${TCL_LIBRARY}

```

snoe925 replied:
Change to a static library.

Commit:
Change to static library.

cpptcl_runtime is now static library.
Add missing PIC flags to link.  No effect for static, but forgetting that
is a hassle.
